src/main/scala/latis/service/hapi/DataService.scala

```diff
@@ -25,33 +26,44 @@ class DataService[F[_]: Concurrent](
   val service: HttpRoutes[F] =
     HttpRoutes.of[F] {
       case GET -> Root / "data"
-          :? IdMatcher(_id)
+          :? DatasetMatcher(_dataset)
+          +& IdMatcher(_id)
+          +& StartTimeMatcher(_startTime)
+          +& StopTimeMatcher(_stopTime)
           +& MinTimeMatcher(_minTime)
           +& MaxTimeMatcher(_maxTime)
           +& ParamMatcher(_params)
           +& IncludeMatcher(_inc)
           +& FormatMatcher(_fmt) =>
-        val req: Either[Status, DataRequest] =
+        val req: Either[Status, DataRequest] = {
+          // For backwards compatibility, allow both
+          // "dataset" or "id",
+          // "start" or "time.min",
+          // "stop" or "time.max"
           for {
-            id      <- _id.asRight
-            minTime <- _minTime.leftMap(_ => Status.`1402`).toEither
-            maxTime <- _maxTime.leftMap(_ => Status.`1403`).toEither
-            _       <- Either.cond(minTime.isBefore(maxTime), (), Status.`1404`)
```

Comment:
This is just a thought, not something that needs to be addressed in this PR.

This line is a great example of validating (as discussed in [Parse, don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)). It checks that `min < max`, fails if it's not, but adds no new information if it is. That means we only know by convention that `min < max`. This check could accidentally be removed or the time values could be changed after the check and we wouldn't know until something went wrong.

If we wanted to parse instead of validate, we could introduce a special pair type `(a, b)` with the invariant that `a < b`. Anywhere we have a value of that special pair type, we also have proof that the start time is before the end time. We couldn't remove this check because it would cause a type error, and we couldn't change the times in a way that violates the invariant.

Reply:
This would be a very handy type. Unfortunately, neither scala's `Range` nor cats' `Range` enforced this. 😞 
Do you know of a handy implementation or do we need to make our own? We might want one that works with our `Data` 
and its `Ordering`. (Later.)

Reply:
I'm not aware of a handy implementation, but it's not hard to make our own:

```scala
sealed abstract case class Lt[A](a: A, b: A)

object Lt {
  def apply[A](a: A, b: A)(implicit ord: Ordering[A]): Option[Lt[A]] =
    if (ord.lt(a, b)) Option(new Lt(a, b) {}) else None
}
```